Source/FieldSolver/SpectralSolver/SpectralFieldData.cpp

```diff
@@ -73,28 +82,44 @@ SpectralFieldData::SpectralFieldData( const amrex::BoxArray& realspace_ba,
         cufftResult result;
 #if (AMREX_SPACEDIM == 3)
         result = cufftPlan3d( &forward_plan[mfi], fft_size[2],
+#    ifdef AMREX_USE_FLOAT
+                              fft_size[1],fft_size[0], CUFFT_R2C);
+#    else
                               fft_size[1],fft_size[0], CUFFT_D2Z);
+#    endif
         if ( result != CUFFT_SUCCESS ) {
             amrex::Print() << " cufftplan3d forward failed! Error: " <<
             cufftErrorToString(result) << "\n";
         }
 
         result = cufftPlan3d( &backward_plan[mfi], fft_size[2],
-                              fft_size[1], fft_size[0], CUFFT_Z2D);
+#    ifdef AMREX_USE_FLOAT
+                              fft_size[1],fft_size[0], CUFFT_C2R);
+#    else
+                              fft_size[1],fft_size[0], CUFFT_Z2D);
+#    endif
```

Comment:
Same as above. Please can you add only the specific float/versus/double part within the #ifdef region.